bin/q

```diff
@@ -780,7 +778,11 @@ class MaterializedFileState(object):
         # multiple people.
         if self.encoding == 'utf-8-sig' and self.lines_read == 0 and not self.skipped_bom:
             try:
-                BOM = self.f.read(3)
+                if six.PY2:
+                    BOM = self.f.read(3)
+                else:
+                    BOM = self.f.buffer.read(3)
+
```

Comment:
You don't like the ternary style?  __BOM = self.f.read(3) if six.PY2 else self.f.buffer.read(3)__